src/utils/network_calculations/ybus_calculations.jl

```diff
@@ -243,9 +246,37 @@ function _goderya(ybus::SparseArrays.SparseMatrixCSC)
     return I
 end
 
-function validate_connectivity(sys::System)
-    a = Adjacency(sys; check_connectivity = false)
-    return validate_connectivity(a.data, collect(get_components(Bus, sys)), a.lookup[1])
+function validate_connectivity(sys::System; method::Union{Nothing, Function} = nothing)
+    buses = sort(
+        collect(get_components(Bus, sys, x -> get_bustype(x) != BusTypes.ISOLATED)),
+        by = b -> get_number(b),
+    )
+    branches = get_components(Branch, sys, get_available)
+
+    preferred_method = length(buses) > 2000 ? find_connected_components : _goderya
+
+    if method === nothing
+        method = preferred_method
+    end
+
+    if method == _goderya
```

Comment:
It seems a little off that you don't actually call this function. Maybe this is internal and doesn't matter, but maybe the user input should be something different, like a string or enum.